packages/arb-token-bridge-ui/src/util/deposits/helpers.ts

```diff
@@ -110,14 +115,78 @@ export const updateAdditionalDepositData = async ({
     })
   }
 
-  // finally, else if the transaction is not ETH ie. it's a ERC20 token deposit
-  return updateTokenDepositStatusData({
+  let { value2 } = depositTx
+
+  // value2 not found in local storage, we fetch it instead
+  if (!value2) {
```

Comment:
This function is originally quite nicely structured. We should try and move this snippet into it's dedicated function like `{ value2 } = getBatchedTransferDepositData(...)` or something similar to enhance readability.